doc/decisions/error_message.md

```diff
@@ -65,6 +65,17 @@ Various FLOSS projects:
     Etcd's approach for errors are tightly coupled to the programming language Go as well as the [gRPC](https://grpc.io/) standard which currently has 
     [16 codes](https://godoc.org/google.golang.org/grpc/codes) defined. Some of these errors are similar or identical to those which will be used in elektra.
     Every error of etcd is associated with one of these categories and gets its own error message which is specified in [this](https://github.com/etcd-io/etcd/blob/master/etcdserver/api/v3rpc/rpctypes/error.go) file. This concept though does not allow easy subclassing which might be useful (eg. further split FailedPrecondition into more specific errors like semantic and syntactic errors)
+* [Windows Registry](https://docs.microsoft.com/en-us/windows/desktop/sysinfo/registry):
+    The registry does not use any specific error concept but takes the standard [Win32 Error Codes](https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-erref/18d8fbe8-a967-4f1c-ae50-99ca8e491d2d). These are neither hierarchical nor have any special ordering. Basically it is the same as elektra has now except for no duplicated
+    errors.
+* Mac OS X plist:
+    Just like Windows, plist uses standard Mac OS X errors which is a [huge catalog](http://krypted.com/lists/comprehensive-list-of-mac-os-x-error-codes/) of unordered
+    return codes as integers.
+* [SNMP Standard](http://www.snmp.com/protocol/):
+    Being a standard network protocol, error codes are very specific to the domain itself. A list can be found [here](https://docs.microsoft.com/en-us/windows/desktop/snmp/snmp-error-codes) and would not meet the needs of elektra at all.
```

Comment:
Where do you see that the error codes are specific to the domain? With a few exceptions (e.g. SNMP_ERRORSTATUS_AUTHORIZATIONERROR) I think most of the error codes have some equivalent in Elektra.

Reply:
UNDOFAILED, WRONGLENGTH, TOOBIG, WRONGENCODING, INCONSISTENTNAME, ... they might all could be mapped to some elektra class but in my opinion these are completely different domains. While SNMP emphasizes on errors in packets, elektra does on handling key value pairs.